paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSinkBuilder.java

```diff
@@ -75,7 +77,7 @@ public class FlinkSinkBuilder {
 
     private DataStream<RowData> input;
     @Nullable protected Map<String, String> overwritePartition;
-    @Nullable protected Integer parallelism;
+    @Nullable private Integer parallelism;
```

Comment:
Could change Integer to int.

Reply:
Different from an Integer, an int cannot be null. The Integer being null or not shows whether the parallelism has been explicitly configured, which might further influences internal implementations. Thus I prefer to keep the existing usage of Integer to better preserve this information.

A similar pattern exists in flink's Transformation, where it needs `boolean parallelismConfigured` along with the `int parallelism`.